### PR TITLE
5patchreviews

### PR DESCRIPTION
**Pull request by Asileka**

### CODE REVIEW

**Review by Angaca — CHANGES_REQUESTED** [review]
Tests need some adjustment. Please take in consideration the following: nonValidId, no body sent, body there but without inc_votes / a totally different key, inc_votes is there but is not a valid integer

File: __tests__/ncgamesapp.test.js [hunk 1]
```diff
@@ -67,3 +67,77 @@ describe("GET:/api/reviews/:review_id", () => {
...
+      review_body: "Farmyard fun!",
+      review_img_url:
+        "https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png",
+      created_at: "2021-01-18T10:00:20.514Z",
+      votes: 2,
+    };
+    return request(app)
+      .patch("/api/reviews/1")
+      .send(newVote)
+      .expect(202)
```
Comment (douglashellowell) [inline __tests__/ncgamesapp.test.js]: a `202` means the request has been accepted but doesn't mean that the server has done anything about it!

a `200` might be better here

[read more](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202)

Comment (Asileka) [reply __tests__/ncgamesapp.test.js]: changed!

File: models/ncgamesmodel.js [hunk 2]
```diff
@@ -17,3 +17,28 @@ exports.fetchReviewByID = (reviewID) => {
       });
   }
 };
+exports.patchReview = (reviewID, newVote) => {
+  let valuesArr = [];
+  if (reviewID) {
+    valuesArr.push(reviewID);
+    return connection
+      .query("SELECT * FROM reviews WHERE review_id = $1", valuesArr)
+      .then((results) => {
```
Comment (douglashellowell) [inline models/ncgamesmodel.js]: you don't need to be doing an extra `SELECT` here 

you can use data in the database (i.e `votes`) in an `UPDATE`!


> if we were trying to add to a column in a partys table
```sql
UPDATE partys
SET attendees = attendees + 1
WHERE party_id = 3;
```

**Review by douglashellowell — CHANGES_REQUESTED** [review]
great tests!

You don't need the extra `SELECT` in your `patchReview` model

have a look at my comment and see if you can refactor to make just one `UPDATE` statement!

could you put in another `/nhelp pr` when fixed please?

**Comment by Asileka** [comment]
Will do, thank you!

**Review by douglashellowell — APPROVED** [review]
nice refactor!

thanks for that

feel free to merge!

**Pull request merged**